### PR TITLE
Fix Maestro Domestic CC detection pattern to prevent false positives

### PR DESCRIPTION
**Pull request by rhoerr**
### Description (*)
The current Maestro Domestic CC detection pattern regex is misformed, causing false detections of any CC number containing `676770` or `676774` as MD type in addition to whatever its proper type is (like VI or AE). This breaks checkout for any customer with a CC number happening to contain that number sequence.

This is easily fixed by wrapping the entire MD pattern in `()` such that the ^ (start) constraint applies to all three clauses, not only the first one.

Of three instances of the MD pattern in core code, two are affected by this issue, the third is not.

### Fixed Issues (if relevant)
1. Fixes [magento/magento2#36332](https://github.com/magento/magento2/issues/36332)

### Manual testing scenarios (*)
1. Go to checkout
2. Select payment method that uses Magento native type detection and CC validation
3. Enter card number "4111 6767 7011 1115"
Note few or no core payment methods use this out of box -- but the validation code is in core nonetheless.

### Questions or comments
The existing REGEX pattern is clearly bugged. This fixes it.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: lib/web/mage/validation.js [hunk 1]
```diff
@@ -138,7 +138,7 @@ define([
             true
         ],
         'MI': [new RegExp('^(5(0|[6-9])|63|67(?!59|6770|6774))\\d*$'), new RegExp('^[0-9]{3}$'), true],
-        'MD': [new RegExp('^6759(?!24|38|40|6[3-9]|70|76)|676770|676774\\d*$'), new RegExp('^[0-9]{3}$'), true]
+        'MD': [new RegExp('^(6759(?!24|38|40|6[3-9]|70|76)|676770|676774)\\d*$'), new RegExp('^[0-9]{3}$'), true]
```
Comment (damienwebdev) [inline lib/web/mage/validation.js:141]: Can you add a test?

File: app/code/Magento/Payment/view/base/web/js/model/credit-card-validation/credit-card-number-validator/credit-card-type.js [hunk 2]
```diff
@@ -103,7 +103,7 @@ define([
         {
             title: 'Maestro Domestic',
             type: 'MD',
-            pattern: '^6759(?!24|38|40|6[3-9]|70|76)|676770|676774\\d*$',
+            pattern: '^(6759(?!24|38|40|6[3-9]|70|76)|676770|676774)\\d*$',
```
Comment (damienwebdev) [inline app/code/Magento/Payment/view/base/web/js/model/credit-card-validation/credit-card-number-validator/credit-card-type.js:106]: Can you add a test?

**Review by damienwebdev — DISMISSED** [review]

File: app/code/Magento/Payment/view/base/web/js/model/credit-card-validation/credit-card-number-validator/credit-card-type.js (hunk 2 above)
Comment (rhoerr) [reply app/code/Magento/Payment/view/base/web/js/model/credit-card-validation/credit-card-number-validator/credit-card-type.js:106]: There wasn't existing coverage of this type validator, but I added a check for it on `Magento_Payment/js/model/credit-card-validation/validator`.

I'm new to it, so pardon any broken conventions.

**Review by Vinai — APPROVED** [review]

**Comment by Vinai** [comment]
The unit test failures are unrelated to this PR.

**Pull request merged**